components/input-mapper/src/main/java/org/wso2/siddhi/extension/input/mapper/wso2event/WSO2SourceMapper.java

```diff
@@ -0,0 +1,224 @@
+/*
+ * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.siddhi.extension.input.mapper.wso2event;
+
+import org.apache.log4j.Logger;
+import org.wso2.siddhi.annotation.Example;
+import org.wso2.siddhi.annotation.Extension;
+import org.wso2.siddhi.annotation.Parameter;
+import org.wso2.siddhi.annotation.util.DataType;
+import org.wso2.siddhi.core.event.Event;
+import org.wso2.siddhi.core.stream.AttributeMapping;
+import org.wso2.siddhi.core.stream.input.InputEventHandler;
+import org.wso2.siddhi.core.stream.input.source.SourceMapper;
+import org.wso2.siddhi.core.util.config.ConfigReader;
+import org.wso2.siddhi.core.util.transport.OptionHolder;
+import org.wso2.siddhi.query.api.definition.Attribute;
+import org.wso2.siddhi.query.api.definition.StreamDefinition;
+import org.wso2.siddhi.query.api.exception.ExecutionPlanValidationException;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+
+/**
+ * This mapper converts WSO2 input event to {@link org.wso2.siddhi.core.event.ComplexEventChunk}. This extension
+ * accepts WSO2 events from WSO2 receiver and maps attribute values to the defined stream.
+ */
+@Extension(
+        name = "wso2event",
+        namespace = "sourceMapper",
+        description = "WSO2 event to Siddhi Event input mapper. Transports which accepts WSO2 messages can utilize " +
+                "this extension to convert the incoming WSO2 message to Siddhi event. Users can send a WSO2 message" +
+                "which should contain the mapping stream attributes in the same order as the defined stream. This " +
+                "conversion will happen without any configs.",
+        parameters = {
+                @Parameter(name = "wso2event.stream.id",
+                        description =
+                                "Used to provide the incoming event's wso2event stream id",
```

Comment:
I want the format to be in the description <name>:<version> and version needs to be in x.x.x format.